### PR TITLE
APIMF 3037

### PR DESCRIPTION
**Pull request by tomsfernandez**
Important commits:
- Fixed semantic ...
- APIMF-3037....
- Test ...

This PR is built on top of: #270 as it has some fixes necessary to fix annotation mapping referencing. Therefore it should be rebased when #270 is approved and merged afterwards.

### CODE REVIEW

File: amf-aml/shared/src/main/scala/amf/plugins/document/vocabularies/model/domain/AnnotationMapping.scala [hunk 1]
```diff
@@ -1,20 +1,31 @@
...
 import amf.plugins.document.vocabularies.metamodel.domain.AnnotationMappingModel
 import amf.plugins.document.vocabularies.metamodel.domain.AnnotationMappingModel._
 import org.yaml.model.YMap
 
 class AnnotationMapping(override val fields: Fields, override val annotations: Annotations)
-    extends PropertyLikeMapping[AnnotationMappingModel.type] {
+    extends PropertyLikeMapping[AnnotationMappingModel.type]
+    with NodeMappable[AnnotationMappingModel.type] {
+
+  override def name: StrField                    = super[NodeMappable].name
```
Comment (nschejtman) [inline amf-aml/shared/src/main/scala/amf/plugins/document/vocabularies/model/domain/AnnotationMapping.scala:14]: Why overriding these methods?

File: amf-aml/shared/src/main/scala/amf/plugins/document/vocabularies/model/domain/AnnotationMapping.scala [hunk 2]
```diff
@@ -1,20 +1,31 @@
...
   override def meta: AnnotationMappingModel.type = AnnotationMappingModel
 
   /** Value , path + field value that is used to compose the id when the object its adopted */
-  override def componentId: String = s"annotation-mappings/${name().value()}"
+  override def componentId: String = s"annotation-mappings/${name.value()}"
+
+  override def linkCopy(): Linkable = AnnotationMapping().withId(id)
+
+  /** apply method for create a new instance with fields and annotations. Aux method for copy */
+  override protected def classConstructor: (Fields, Annotations) => Linkable with DomainElement = ???
```
Comment (nschejtman) [inline amf-aml/shared/src/main/scala/amf/plugins/document/vocabularies/model/domain/AnnotationMapping.scala:28]: Add TODO

File: amf-aml/shared/src/main/scala/amf/plugins/document/vocabularies/model/domain/SemanticExtension.scala [hunk 3]
```diff
@@ -10,11 +10,11 @@ import org.yaml.model.YMap
...
   override def meta: SemanticExtensionModel.type = SemanticExtensionModel
 
-  def extensionName(): StrField                       = fields.field(ExtensionName)
-  def extensionMappingDefinition(): AnnotationMapping = fields.field(ExtensionMappingDefinition)
+  def extensionName(): StrField              = fields.field(ExtensionName)
+  def extensionMappingDefinition(): StrField = fields.field(ExtensionMappingDefinition)
 
   def withExtensionName(name: String): SemanticExtension = set(ExtensionName, name)
-  def withExtensionMappingDefinition(annotationMapping: AnnotationMapping): SemanticExtension =
+  def withExtensionMappingDefinition(annotationMapping: String): SemanticExtension =
```
Comment (nschejtman) [inline amf-aml/shared/src/main/scala/amf/plugins/document/vocabularies/model/domain/SemanticExtension.scala:17]: Guessing you moved this to strings to checkNodeMappableReferences

File: amf-aml/shared/src/main/scala/amf/plugins/document/vocabularies/model/domain/AnnotationMapping.scala (hunk 1 above)
Comment (tomsfernandez) [reply amf-aml/shared/src/main/scala/amf/plugins/document/vocabularies/model/domain/AnnotationMapping.scala:14]: I had to add NodeMappable to the class hierarchy. Both PropertyLikeMapping and NodeMappable implement the `name` and `withName` methods so I had to override them in the extending class.

File: amf-aml/shared/src/main/scala/amf/plugins/document/vocabularies/model/domain/AnnotationMapping.scala (hunk 2 above)
Comment (tomsfernandez) [reply amf-aml/shared/src/main/scala/amf/plugins/document/vocabularies/model/domain/AnnotationMapping.scala:28]: Added implementation

File: amf-aml/shared/src/main/scala/amf/plugins/document/vocabularies/model/domain/SemanticExtension.scala (hunk 3 above)
Comment (tomsfernandez) [reply amf-aml/shared/src/main/scala/amf/plugins/document/vocabularies/model/domain/SemanticExtension.scala:17]: Yes, I made AnnotationMapping extend NodeMappable to do this.

**Review by nschejtman — APPROVED** [review]

**Pull request merged**